pdf/pdflinkarea.cpp

```diff
@@ -148,8 +192,50 @@ void PDFLinkArea::mouseReleaseEvent(QMouseEvent *event)
     if (!rect.contains(event->pos()))
         return;
 
+    // Click action logic in order.
+    // - click on selection;
+    // - unselect if selection is set;
+    // - click on annotation;
+    // - click on link;
+    // - click.
+
     if (!d->canvas) {
         emit clicked(d->clickLocation);
+    } else if (d->clickOnSelection) {
+        emit selectionClicked();
+    } else if (d->selection && d->selection->count() > 0) {
+        d->selection->unselect();
+    } else if (d->annotation != nullptr) {
+        // proxy will be child of this object to avoid memory leak.
```

Comment:
I've added the following comment:
>  Todo: transfer ownership to QML, if possible. Before that,
     all created proxy objects will be alive up to the moment
     the document object is released.